src/net/sourceforge/kolmafia/session/ChoiceAdventures.java

```diff
@@ -8969,10 +8969,209 @@ else if (KoLCharacter.isEd()) {
+  }
+
+  static {
+    addAdjectives(
+        ShadowTheme.FIRE,
+        "steaming",
+        "scalded",
+        "scalding",
+        "flame-choked",
+        "blazing",
+        "ember-lit",
+        "scorching",
+        "blistering",
+        "sizzling",
+        "charred",
+        "white-hot");
+    addAdjectives(
+        ShadowTheme.MATH,
+        "Pythagorean",
+        "exponential",
+        "fractional",
+        "parabolic",
+        "sinusoidal",
+        "monomial",
+        "boolean",
+        "integer",
+        "hyperbolic",
+        "Riemannian",
+        "irrational",
+        "geometric",
+        "fractal",
+        "ordinal",
+        "logarithmic",
+        "angular",
+        "Cartesian",
+        "prime",
+        "multiplicative",
+        "Fibonacci",
+        "self-referential",
+        "trigonometric",
+        "cubic",
+        "vector");
+    addAdjectives(
+        ShadowTheme.WATER,
+        "humid",
+        "dripping",
+        "water-logged",
+        "runny",
+        "soaked",
+        "drowning",
+        "watery",
+        "aqueous",
+        "sodden",
+        "moist",
+        "foggy",
+        "damp");
+    addAdjectives(
+        ShadowTheme.TIME,
+        "rickety",
+        "ramshackle",
+        "old",
+        "broken-down",
+        "ancient",
+        "derelict",
+        "shabby",
+        "crumbling",
+        "ruined",
+        "decaying",
+        "antique",
+        "dilapidated",
+        "unkempt");
+    addAdjectives(
+        ShadowTheme.BLOOD,
+        "bloodstained",
+        "vein-shot",
+        "blood-soaked",
+        "blood-drenched",
+        "crimson",
+        "pulsing",
+        "sanguine",
+        "veiny",
+        "bloody",
+        "hematic",
+        "bleeding");
+    addAdjectives(
+        ShadowTheme.COLD,
+        "frost-rimed",
+        "icy",
+        "iced-over",
+        "frozen",
+        "freezing",
+        "frosty",
+        "spectral",
+        "cold-numbed",
+        "snow-covered",
+        "chilly",
+        "hyperborean",
+        "wintry",
+        "arctic",
+        "frigid");
+    addAdjectives(
+        ShadowTheme.GHOST,
+        "ghostly",
+        "insubstantial",
+        "diaphanous",
+        "translucent",
+        "see-through",
+        "gossamer",
+        "half-there",
+        "nearly invisible",
+        "wispy",
+        "ephemeral");
+  }
+
+  private static final Pattern ADJECTIVE_PATTERN = Pattern.compile(" the ([^ ]+) ");
+
+  public static ShadowTheme shadowLabyrinthTheme(String text) {
+    Matcher m = ADJECTIVE_PATTERN.matcher(text);
+    return m.find() ? adjectiveToTheme.get(m.group(1)) : null;
+  }
+
+  public static ChoiceOption shadowLabyrinthSpoiler(String text) {
+    ShadowTheme theme = shadowLabyrinthTheme(text);
+    String spoiler = "unknown theme";
+    if (theme != null) {
+      String normal = theme.getNormal();
+      spoiler = normal;
+      String artifact = theme.getArtifact();
+      if (artifact != null) {
```

Comment:
I believe the artifact is received if and only if you're currently on a quest for Rufus to receive it. We can parse that from the initial choice (or quest log?), store it in a preference, and look it up here.

I don't know whether it's worth showing that you could receive the other artifacts. If we only show the one you're interested in, that makes it clear which choice you should pick to complete the quest.

Reply:
This is feature creep, though, so it's fine as-is if you want to get it in.

Reply:
I agree we will want to do that: if you are on a quest for an artifact, show only that option. Otherwise, show the normal result.

I figure adding the tracking will come in another PR and we can adjust this then.